### PR TITLE
Feat(opsgenie): deprecate Opsgenie plugin

### PR DESCRIPTION
**Pull request by mifu67**
Hide the Opsgenie plugin if it's not installed. If it is, mark it as legacy.

Fixes ER-1790

### CODE REVIEW

**Review by dashed — APPROVED** [review]

**Review by EricHasegawa — APPROVED** [review]
👀

**Review by leeandher — COMMENTED** [review]
Just an FYI, is this the expected behavior?

File: src/sentry/plugins/__init__.py [hunk 1]
```diff
@@ -5,4 +5,5 @@
     "slack",
     "jira",
     "pagerduty",
+    "opsgenie",
```
[the remark this answers is not in the document]
Comment (mifu67) [reply src/sentry/plugins/__init__.py:8]: I think that's what Dhrumil was asking for, but I'll double check.

[the remark this answers is not in the document]
Comment (mifu67) [reply src/sentry/plugins/__init__.py:8]: Ah, I was wrong. He was just looking to hide the plugin for orgs who have opted in. Not merging this until GA.

**Pull request merged**